src/linear_algebra/affine_space.lean

```diff
@@ -605,48 +595,48 @@ end
 /-- If a family is affinely independent, so is any subfamily indexed
 by a subtype of the index type. -/
 lemma affine_independent_subtype_of_affine_independent {p : ι → P}
-    (ha : affine_independent k V p) (s : set ι) : affine_independent k V (λ i : s, p i) :=
+    (ha : affine_independent k p) (s : set ι) : affine_independent k (λ i : s, p i) :=
 affine_independent_embedding_of_affine_independent (function.embedding.subtype _) ha
 
 end affine_independent
 
-namespace affine_space
+section affine_space'
 
-variables (k : Type*) (V : Type*) (P : Type*) [ring k] [add_comm_group V] [module k V]
-variables [affine_space k V P]
+variables (k : Type*) {V : Type*} (P : Type*) [ring k] [add_comm_group V] [module k V]
+variables [affine_space V P]
+include V
 
 /-- A `simplex k V P n` is a collection of `n + 1` affinely
 independent points. -/
 structure simplex (n : ℕ) :=
 (points : fin (n + 1) → P)
-(independent : affine_independent k V points)
+(independent : affine_independent k points)
 
-/-- A `triangle k V P` is a collection of three affinely independent
-points. -/
-abbreviation triangle := simplex k V P 2
+/-- A `triangle k V P` is a collection of three affinely independent points. -/
+abbreviation triangle := simplex k P 2
```

Comment:
This is the place where I noted on Zulip that it seems desirable to use a namespace rather than claiming `simplex` and `triangle` in the root namespace for these particular concepts among various mathematical definitions that may use those names.

Reply:
Also, the doc strings referring to `simplex k V P` and `triangle k V P` definitely need updating for `V` no longer being an explicit argument. Likewise all other doc strings that name a `V` argument explicitly where it's now implicit.

Reply:
I updated those doc strings. The concern about these particular definitions still needing a namespace (whether `affine_space`, `affine` or something else along those lines) still applies.

Reply:
I've now put in the namespace `affine` (I think the use of `affine_space` as `notation` prevents it being used as a namespace as well).